services/src/modules/authentication/get-secret.ts

```diff
@@ -0,0 +1,49 @@
+import fastify from 'fastify';
+
+// There is problem with typings in the module: https://github.com/auth0/node-jwks-rsa/issues/78
+import * as jwks from 'jwks-rsa';
+import { ClientOptions, JwksClient } from 'jwks-rsa';
+import { decode } from 'jsonwebtoken';
+import logger from '../logger';
+import { authenticationConfig } from '../config';
+
+type DecodeJWT = {
+  header: Record<string, unknown>;
+  payload: Record<string, unknown>;
+  signature: string;
+};
+
+const createClient = (jwks as unknown) as (options: ClientOptions) => JwksClient;
+
+export default function getSecret(
+  request: fastify.FastifyRequest,
+  _reply: fastify.FastifyReply<unknown>,
+  cb: (e: Error | null, secret: string | undefined) => void
+): void {
+  const authHeader: string = request.headers.authorization;
+  const token = authHeader.split(' ')[1];
+  const decodedToken = decode(token, { complete: true, json: true }) as DecodeJWT;
```

Comment:
Can't we somehow re-use the token in the [request context](https://github.com/Soluto/stitch/blob/master/services/src/modules/gateway.ts#L27) (or use the one decoded here in there) instead of decoding it twice in both places?